Added support for custom actions and filters

carlalexander opened the pull request:
First pass at #3. This adds the ability to create custom actions and filters using dictionaries. This should replace the need for a lot of the files that are in the role. 

For example, `owncloud.local` should be generated by `debops.owncloud`. I didn't remove it yet, but that's something that could be done once the PR gets merged.

carlalexander committed:
Added support for custom local actions and filters

carlalexander commented:
Found a typo running some tests. Don't merge yet. Will update PR.

carlalexander commented:
Fixed the issue :)

drybjed commented:
Very nice. Although, do you think that you could move the template documentation into the `docs/defaults-configuration.rst` file? That way you can also provide some examples how to use the templates, and everything is in one place.

carlalexander commented:
Sure, I'll rework the documentation. :smile:

carlalexander commented:
Alright! Updated the PR. I added everything to `docs/defaults-configuration.rst`, but kept the doc in the templates as well. I find it useful in `debops.nginx`. Do you prefer I take it out?

I also added some more configuration options and I found a few bugs during testing.

drybjed commented on docs/defaults-configuration.rst:
```diff
@@ -9,6 +9,81 @@ them.
+  with ``fail2ban`` user rights.
+
+``check``
+  Optional. Command executed once before each ``ban` command.
```
It should be:

```
``ban``
```

carlalexander replied:
Oups. Fixing that!

drybjed commented:
@carlalexander Thanks! The documentation in templates was an old idea I played with but it seems that it's less accessible than documentation using RTFD, which can be searchable and you can provide examples easily. So I want to remove the documentation in `debops.nginx` and switch it to RTFD at some point. IMO you can remove the duplicate docs in the templates, but I won't hold it against you. :-)

carlalexander commented:
Here you go! I fixed the typo and removed the documentation in the templates. :smile:

drybjed commented:
Great, thank you. I'll update the documentation and make a new release on the weekend.
